### PR TITLE
Gentx init

### PR DESCRIPTION
**Pull request by joneskm**

### CODE REVIEW

File: x/staking/src/keeper/mod.rs [hunk 1]
```diff
@@ -442,12 +442,12 @@ impl<
             match validator.status {
                 BondStatus::Unbonded => {
                     self.unbonded_to_bonded(ctx, &mut validator)?;
-                    amt_from_bonded_to_not_bonded =
+                    amt_from_not_bonded_to_bonded =
```
Comment (dmvict) [inline x/staking/src/keeper/mod.rs:445]: You may use shorter syntax. Clippy fix.

File: tendermint/src/types/proto/validator.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 use address::ValAddress;
+use ibc_proto::protobuf::Protobuf;
```
Comment (dmvict) [inline tendermint/src/types/proto/validator.rs:2]: Does it differ to `core::Protobuf`?

File: gaia-rs/scripts/init.sh [hunk 3]
```diff
@@ -10,4 +10,12 @@ set -eux
 rm -rf ~/.gaia-rs
 cargo run -- init test
 
+echo "Generating deterministic account - alice"
+echo "race draft rival universe maid cheese steel logic crowd fork comic easy truth drift tomorrow eye buddy head time cash swing swift midnight borrow" | gaia-rs keys add alice --recover --keyring-backend=test
+
 cargo run -- add-genesis-account cosmos1syavy2npfyt9tcncdtsdzf7kny9lh777pahuux "$amount"uatom
+
+cargo run -- gentx 10000000000uatom --from-key alice --min-self-delegation 1 --moniker test --account-number 0 --sequence 0 --keyring-backend=test
```
Comment (NYBACHOK) [inline gaia-rs/scripts/init.sh:18]: Why sequence is zero? When I tested it required 2

File: docs/tutorials/staking.md [hunk 4]
```diff
@@ -1,8 +1,10 @@
 # Staking tutorial
 
+## Adding validators
+
 1. Create a validator
 
-```
+```shell
 gaiad tx staking create-validator \
```
[the remark this answers is not in the document]
Comment (joneskm) [reply docs/tutorials/staking.md]: I've deleted that file completely - it was an old tutorial.

File: gaia-rs/scripts/init.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (joneskm) [reply gaia-rs/scripts/init.sh:18]: Even in a simple bash script?

File: x/staking/src/keeper/mod.rs (hunk 1 above)
Comment (joneskm) [reply x/staking/src/keeper/mod.rs]: Fixed.

File: tendermint/src/types/proto/validator.rs (hunk 2 above)
Comment (joneskm) [reply tendermint/src/types/proto/validator.rs:2]: As in the one in Gears? I don't think so, but this crate doesn't depend on Gears.

File: gaia-rs/scripts/init.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmvict) [reply gaia-rs/scripts/init.sh:18]: Lets imagine that someone tries to make genesis account with amount of tokens < 10000000000uatom

The command should fail in this case.

File: tendermint/src/types/proto/validator.rs (hunk 2 above)
Comment (dmvict) [reply tendermint/src/types/proto/validator.rs:2]: Oh, my bad. I didn't noticed that.

File: gears/src/x/ante.rs [hunk 5]
```diff
@@ -462,13 +461,20 @@ impl<
                 .get_public_key()
                 .expect("account pub keys are set in set_pub_key_ante_handler"); //TODO: but can't they be set to None?
 
+            let genesis = ctx.height() == 0;
+            let account_number = if genesis {
+                0
```
Comment (NYBACHOK) [inline gears/src/x/ante.rs:466]: Ah. So that's why you are able to pass `account_number`=0

File: x/genutil/src/gentx.rs [hunk 6]
```diff
@@ -170,21 +170,12 @@ impl TxHandler for GentxTxHandler {
...
-                        Ed25519PubKey::try_from(var.verification_key().as_bytes().to_vec())?,
-                    ),
-                    _ => unreachable!(),
-                }
+                let file = std::fs::File::open(
+                    client_tx_context
+                        .home
+                        .join("config/priv_validator_key.json"), // TODO: delegate this to tendermint crate
+                )?;
+                tendermint::get_validator_pub_key(file).unwrap().into()
```
Comment (NYBACHOK) [inline x/genutil/src/gentx.rs:178]: Unwrap? Maybe return error, but place todo?

**Review by NYBACHOK — APPROVED** [review]

File: x/genutil/src/gentx.rs (hunk 6 above)
Comment (joneskm) [reply x/genutil/src/gentx.rs]: Ooops forgot to fix that... fixed

File: gaia-rs/scripts/init.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (joneskm) [reply gaia-rs/scripts/init.sh:18]: Fair point... do we actually need the amount to be passed as an argument? Do you use that?

[the remark this answers is not in the document]
Comment (joneskm) [reply gaia-rs/scripts/init.sh:13]: Done.

**Review by dmvict — APPROVED** [review]

**Pull request merged**